v2/apiserver/internal/core/events.go

```diff
@@ -290,6 +301,11 @@ type EventsService interface {
 		context.Context,
 		EventsSelector,
 	) (DeleteManyEventsResult, error)
+	// Retry copies an Event, including Worker configuration and Jobs, and
+	// creates a new Event from this information.  Where possible, jobs are not
+	// re-scheduled and their results re-used, for example when a job has
+	// succeeded and does not make use of a shared workspace.
+	Retry(context.Context, string) (Event, error)
```

Comment:
Nit: The word "cache" could be misleading. I wonder if it might be more accurate to say we "inherit" results where possible?